### PR TITLE
Update minimum Python versions

### PR DESCRIPTION
**Pull request by hugovk**
And fix some typos.

### CODE REVIEW

File: doc/users/installing.rst [hunk 1]
```diff
@@ -10,20 +10,20 @@ Dependencies
 **Requirements**
 
 These are external packages which you will need to install before
-installing basemap. 
+installing Basemap. 
 
 
-matplotlib 1.0.0 (or later, `download <http://sf.net/projects/matplotlib/>`__)
+Matplotlib 1.0.0 (or later, `download <http://sf.net/projects/matplotlib/>`__)
```
Comment (WeatherGod) [inline doc/users/installing.rst]: heh, probably should update that to the matplotlib.org address.

File: doc/users/installing.rst [hunk 2]
```diff
@@ -10,20 +10,20 @@ Dependencies
...
 
-Python 2.4 (or later, including Python 3)
-    matplotlib requires python 2.4 or later (`download <http://www.python.org/download/>`__)
+Python 2.6 (or later, including Python 3)
+    Matplotlib requires Python 3.5 or later (`download <http://www.python.org/download/>`__)
```
Comment (WeatherGod) [inline doc/users/installing.rst]: Matplotlib 2.2.x will be an LTS branch, and supports py27. Matplotlib v3 hasn't been released yet, which will support a minimum of py35, yes. Should probably think of how to phrase this.

File: doc/users/installing.rst [hunk 3]
```diff
@@ -10,20 +10,20 @@ Dependencies
...
 
-numpy 1.2.1 (or later)
-    array support for python (`download <http://sourceforge.net/project/showfiles.php?group_id=1369&package_id=175103>`__)
+NumPy 1.2.1 (or later)
+    Array support for Python (`download <http://sourceforge.net/project/showfiles.php?group_id=1369&package_id=175103>`__)
```
Comment (WeatherGod) [inline doc/users/installing.rst]: should probably update this link, too.

File: doc/users/installing.rst (hunk 2 above)
Comment (hugovk) [reply doc/users/installing.rst]: How's the latest commit look?

**Pull request merged**